src/clr-addons/datagrid/datagrid-state-persistence/datagrid-field.directive.ts

```diff
@@ -1,9 +1,15 @@
 import { Directive, Input } from '@angular/core';
 
 @Directive({
-  selector: '[clrDgField]',
+  selector: '[clrDgField],[clrDgFieldKey]',
 })
 export class DatagridFieldDirective {
   @Input()
   clrDgField: string;
+  @Input()
+  clrDgFieldKey: string;
+
+  get persistenceKey(): string {
```

Comment:
could you refactor this getter to a method (getFieldName) pls?
personally I don't like getter with logic inside as this is so magic when you use it (you think it is a property but actually calculates something).

Reply:
Changed it